File: content/code/dart-keyboard-event/arrow-key.dart
import 'dart:html';

void arrowKeystrokes(Event e) {
  if(e is KeyboardEvent) {
    KeyboardEvent kevt = e as KeyboardEvent;

    switch(kevt.keyCode) {
      case KeyCode.LEFT:
        querySelector("#info").appendHtml("LEFT ");
        break;

      case KeyCode.UP:
        querySelector("#info").appendHtml("UP ");
        break;

      case KeyCode.DOWN:
        querySelector("#info").appendHtml("DOWN ");
        break;

      case KeyCode.RIGHT:
        querySelector("#info").appendHtml("RIGHT ");
        break;

      default:
        querySelector("#info").appendHtml("SOMEKEY ");
    }
  }
}

void main() {
  window.onKeyUp.listen(arrowKeystrokes);
}

File: content/code/javascript/keyboard-event/arrow-key.js
var Key = {
  LEFT:   37,
  UP:     38,
  RIGHT:  39,
  DOWN:   40
};

/**
 * old IE: attachEvent
 * Firefox, Chrome, or modern browsers: addEventListener
 */
function _addEventListener(evt, element, fn) {
  if (window.addEventListener) {
    element.addEventListener(evt, fn, false);
  }
  else {
    element.attachEvent('on'+evt, fn);
  }
}

function handleKeyboardEvent(evt) {
  if (!evt) {evt = window.event;} // for old IE compatible
  var keycode = evt.keyCode || evt.which; // also for cross-browser compatible

  var info = document.getElementById("info");
  switch (keycode) {
    case Key.LEFT:
      info.value += "LEFT ";
      break;
    case Key.UP:
      info.value += "UP ";
      break;
    case Key.RIGHT:
      info.value += "RIGHT ";
      break;
    case Key.DOWN:
      info.value += "DOWN ";
      break;
    default:
      info.value += "SOMEKEY ";
  }
}

_addEventListener('keydown', document, handleKeyboardEvent);
